lib/msal-angular/src/msal.interceptor.ts

```diff
@@ -116,36 +117,51 @@ export class MsalInterceptor implements HttpInterceptor {
 
         const matchingProtectedResources = this.matchResourcesToEndpoint(protectedResourcesArray, normalizedEndpoint);
 
-        if (matchingProtectedResources.length > 0) {
-            return this.matchScopesToEndpoint(this.msalInterceptorConfig.protectedResourceMap, matchingProtectedResources, httpMethod);
+        // Check absolute urls of resources first before checking relative to prevent incorrect matching where multiple resources have similar relative urls
+        if (matchingProtectedResources.absoluteResources.length > 0) {
+            return this.matchScopesToEndpoint(this.msalInterceptorConfig.protectedResourceMap, matchingProtectedResources.absoluteResources, httpMethod);
+        } else if (matchingProtectedResources.relativeResources.length > 0){
+            return this.matchScopesToEndpoint(this.msalInterceptorConfig.protectedResourceMap, matchingProtectedResources.relativeResources, httpMethod);
         }
 
         return null;
     }
 
     /**
      * Finds resource endpoints that match request endpoint
-     * @param protectedResourcesArray 
+     * @param protectedResourcesEndpoints
      * @param endpoint 
-     * @param location 
      * @returns 
      */
-    private matchResourcesToEndpoint(protectedResourcesEndpoints: string[], endpoint: string): Array<string> {
-        return protectedResourcesEndpoints.filter(key => {
+    private matchResourcesToEndpoint(protectedResourcesEndpoints: string[], endpoint: string): MatchingResources {
+        const matchingResources: MatchingResources = {absoluteResources: [], relativeResources: []};
+
+        protectedResourcesEndpoints.forEach(key => {
+            // Normalizes and adds resource to matchingResources.absoluteResources if key matches endpoint. StringUtils.matchPattern accounts for wildcards
             const normalizedKey = this.location.normalize(key);
+            if (StringUtils.matchPattern(normalizedKey, endpoint)){
+                matchingResources.absoluteResources.push(key);
+            }
             
+            // Transforms relative urls to absolute urls, gets components
+            const keyLink = document.createElement("a");
+            keyLink.href = key;
+            const keyLinkComponents = new UrlString(keyLink.href).getUrlComponents();
+
+            const endpointLink = document.createElement("a");
+            endpointLink.href = endpoint;
+            const endpointLinkComponents = new UrlString(endpointLink.href).getUrlComponents();
```

Comment:
Can you factor this logic into a helper function?

Reply:
Sure. I had to leave the `new UrlString...` part out of the helper, since the absolute endpoint is needed later on.